src/pyspiffe/svid/jwt_svid_validator.py

```diff
@@ -44,21 +45,21 @@ def __init__(self) -> None:
         pass
 
     def validate_header(self, header: Dict[str, str]) -> None:
```

Comment:
reading the docstring I think that this method should be renamed to `validate_headers` and the dict param to `headers`. Would you do that as part of this PR?

Reply:
Sure. I'll update the PR with that change.
Thanks Max.

Reply:
I don't agree with this change.. The token has 1 header and multiple key-values inside. And here, this method is validating if the header has the required values. The header is provided as input for the method (in the form of Dic) containing the key-value pairs that composes the token's header.

Reply:
The SPIFFE standard refers the parts of the token header as, for example, `alg` header, `typ` header, and as we are validating a couple of them I thought it would make sense to rename it as `validate_headers`. On the other hand, the RFC 7518 only uses `header` as a whole and then talks about header parameters to refer the specific parts. So, either way is ok for me, your call @glaucimar ;)

Reply:
`def validate_header(self, parameters: Dict[str, str]) -> None:` would work here? What do you think?

Reply:
> `def validate_header(self, parameters: Dict[str, str]) -> None:` would work here? What do you think?

Ah.. to me, that makes sense.

Reply:
> The SPIFFE standard refers the parts of the token header as, for example, `alg` header, `typ` header, and as we are validating a couple of them I thought it would make sense to rename it as `validate_headers`. On the other hand, the RFC 7518 only uses `header` as a whole and then talks about header parameters to refer the specific parts. So, either way is ok for me, your call @glaucimar ;)

Thanks @maxlambrecht for bringing all of this information.

Reply:
Sounds good

Reply:
Ok, I made che renamed and also renamed the variable used to pass the parameter to the function based on the docstrings from `jwt.get_unverified_header(token)`:
```
    def get_unverified_header(self, jwt):
        """Returns back the JWT header parameters as a dict()

        Note: The signature is not verified so the header parameters
        should not be fully trusted until signature verification is complete
        """
        headers = self._load(jwt)[2]
        self._validate_headers(headers)

        return headers
```

Also in this new commit I updated some test cases according to a recent change where a parameter (audience) change its type, from List to Set.